tensorboard/webapp/hparams/_redux/hparams_effects.ts

```diff
@@ -0,0 +1,105 @@
+/* Copyright 2023 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+import {Injectable} from '@angular/core';
+import {Actions, createEffect, ofType} from '@ngrx/effects';
+import {Store} from '@ngrx/store';
+import {Observable, of, throwError, merge} from 'rxjs';
+import {
+  catchError,
+  filter,
+  map,
+  switchMap,
+  withLatestFrom,
+  throttleTime,
+  combineLatestWith,
+} from 'rxjs/operators';
+
+import {navigated} from '../../app_routing/actions';
+import {getExperimentIdsFromRoute} from '../../app_routing/store/app_routing_selectors';
+import {State} from '../../app_state';
+import * as coreActions from '../../core/actions';
+import * as runsActions from '../../runs/actions/runs_actions';
+import {HttpErrorResponse} from '../../webapp_data_source/tb_http_client';
+
+import * as hparamsActions from './hparams_actions';
+import {HparamsDataSource} from './hparams_data_source';
+import {HparamAndMetricSpec, SessionGroup} from '../types';
+import {getEnableHparamsInTimeSeries} from '../../feature_flag/store/feature_flag_selectors';
+
+/**
+ * Effects for fetching the hparams data from the backend.
+ */
+@Injectable()
+export class HparamsEffects {
+  constructor(
+    private readonly actions$: Actions,
+    private readonly store: Store<State>,
+    private readonly dataSource: HparamsDataSource
+  ) {}
+
+  private readonly runTableShown$: Observable<string[]> = this.actions$.pipe(
+    ofType(runsActions.runTableShown),
```

Comment:
If I understand correctly, runTableShown is fired when any run table is shown - either in experiment list or in dashboard, right?

If so, then loading any runs table ultimately dispatches an instance of hparamsFetchSessionGroupsSucceeded which populates the same state over and over - overwriting whatever was there before. The action doesn't clarify which run table you've loaded data for - is it one of the tables in the experiment list or is it the table in the dashboard?

Reply:
I'm unsure how we can differentiate between them. This does remind me that I had intended to ensure the events are only dispatched on the dashboard routes. How would you feel about that solution?

Reply:
I think you could encode the information in the action that triggers the logic. For example:
1. have two different actions `runTableShownInExperimentList` and `runTableShownInDashboard` 
2. or make it a property of the `runTableShown` action. Maybe an enum of type `RunTableLocation`.

On the other hand, using route information for this seems ok. But just to confirm: Do you intend that this logic will also handle the run tables in the experiment list? And, if so, do you plan to somehow differentiate the two cases when you dispatch the action after retrieving the data - perhaps add a property to `hparamsFetchSessionGroupsSucceeded` (again maybe an enum type `RunTableLocation`).

Reply:
At the moment I do not intend for this to be active on the experiment list at all. Once this portion of the code is stable I plan to do a refactor of the data fetching logic on the experiment list though and at that point I may try to merge these. If that happens I will probably add an enum to the `runTableShown` action payload.